src/function/scalar/list/list_contains.cpp

```diff
@@ -0,0 +1,221 @@
+
+		result_entries[list_index] = false;
+		for (idx_t child_idx = 0; child_idx < list_entry.length; child_idx++) {
+			auto child_value_idx = source_idx + child_idx;
+			if (!child_data.validity.RowIsValid(child_value_idx)) {
+				continue;
+			}
+			if (ValueCompare<Value>(child_vector.GetValue(child_value_idx), value_vector.GetValue(value_index))) {
+				result_entries[list_index] = true;
+				break; // Found value in list, no need to look further
+			}
+		}
+	}
+}
+
+static void ListContainsFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	switch (args.data[1].GetType().InternalType()) {
+	case PhysicalType::BOOL:
+	case PhysicalType::INT8:
+		TemplatedListContainsFunction<int8_t>(args, state, result);
+		break;
+	case PhysicalType::INT16:
+		TemplatedListContainsFunction<int16_t>(args, state, result);
+		break;
+	case PhysicalType::INT32:
+		TemplatedListContainsFunction<int32_t>(args, state, result);
+		break;
+	case PhysicalType::INT64:
+		TemplatedListContainsFunction<int64_t>(args, state, result);
+		break;
+	case PhysicalType::INT128:
+		TemplatedListContainsFunction<hugeint_t>(args, state, result);
+		break;
+	case PhysicalType::UINT8:
+		TemplatedListContainsFunction<uint8_t>(args, state, result);
+		break;
+	case PhysicalType::UINT16:
+		TemplatedListContainsFunction<uint16_t>(args, state, result);
+		break;
+	case PhysicalType::UINT32:
+		TemplatedListContainsFunction<uint32_t>(args, state, result);
+		break;
+	case PhysicalType::UINT64:
+		TemplatedListContainsFunction<uint64_t>(args, state, result);
+		break;
+	case PhysicalType::FLOAT:
+		TemplatedListContainsFunction<float>(args, state, result);
+		break;
+	case PhysicalType::DOUBLE:
+		TemplatedListContainsFunction<double>(args, state, result);
+		break;
+	case PhysicalType::VARCHAR:
+		TemplatedListContainsFunction<string_t>(args, state, result);
+		break;
+	case PhysicalType::MAP:
+	case PhysicalType::STRUCT:
+	case PhysicalType::LIST:
+		NestedListContainsFunction(args, state, result);
+		break;
+	default:
+		throw NotImplementedException("This function has not been implemented for this type");
+	}
+}
+
+static unique_ptr<FunctionData> ListContainsBind(ClientContext &context, ScalarFunction &bound_function,
+                                                 vector<unique_ptr<Expression>> &arguments) {
+	D_ASSERT(bound_function.arguments.size() == 2);
+
+	const auto &list = arguments[0]->return_type; // change to list
+	const auto &value = arguments[1]->return_type;
+	if (list.id() == LogicalTypeId::SQLNULL && value.id() == LogicalTypeId::SQLNULL) {
+		bound_function.arguments[0] = LogicalType::SQLNULL;
+		bound_function.arguments[1] = LogicalType::SQLNULL;
+		bound_function.return_type = LogicalType::SQLNULL;
+	} else if (list.id() == LogicalTypeId::SQLNULL || value.id() == LogicalTypeId::SQLNULL) {
+		// In case either the list or the value is NULL, return NULL
+		// Similar to behaviour of prestoDB
+		bound_function.arguments[0] = list;
```

Comment:
I'm wondering  should we follow presto's behavior here as NULL seems to be a valid value in a list. ex.
```sql
select list_contains([1, 2, NULL], 1)
```
Will output `true`.
Thus, a user may expect
```sql
select list_contains([1, 2, NULL], NULL)
```
Also to be true.